R/splatPop-simulate.R

```diff
@@ -0,0 +1,849 @@
+    set.seed(getParam(params, "seed"))
+    
+    nGroups <- getParam(params, "nGroups")
+    
+    if (verbose) {message("Sampling population level value....")}
+    vcf.parsed <- splatPopParseVCF(vcf, params)
+    group.names <- paste0("Group", seq_len(nGroups))
+    
+    if(!is.data.frame(key)){key <- splatPopParseGenes(params, gff)}
+    if (!all(c("exp_mean", "exp_cv") %in% names(key))){
+        key <- splatPopAssignMeans(params, key)
+    }
+    
+    if (!all(c("eQTL", "eSNP", "EffectSize") %in% names(key))){
+        key <- splatPopeQTLEffects(params, key, vcf.parsed)
+        
+        if(length(group.names) > 1){
+            key <- splatPopGroupEffects(params, key, group.names)
+            }
+    }
+    
+    if (verbose) {message("Simulating gene means for population...")}
+
+    MeansPop <- splatPopSimMeans(vcf.parsed, key)
+    MeansPop <- splatPopQuantNorm(params, MeansPop)
+    key <- splatPopQuantNormKey(key, MeansPop) 
+    
+    eMeansPop <- splatPopSimEffects('global', key, vcf.parsed, MeansPop)
+    
+    if(length(group.names) > 1){
+        eMeansPopq_groups <- list()
+        for(id in group.names){
+            eMeansPop.g <- splatPopSimEffects(id, key, vcf.parsed, eMeansPop)
+            eMeansPop.g[eMeansPop.g <= 0] <- 1e-5 
+            eMeansPopq_groups[[id]] <- eMeansPop.g}
+        return(list(means=eMeansPopq_groups, key=key))
+        
+    }else{
+        eMeansPop[eMeansPop <= 0] <- 1e-5 
+        return(list(means=eMeansPop, key=key))
+    }
+}
+
+
+#' splatPopSimulateSC
+#'
+#' Simulate count data for a population from a fictional single-cell 
+#' RNA-seq experiment using the Splat method.
+#' 
+#' @param sim_means Matrix or list of matrices of gene means for the population. 
+#'        Output from `splatPopSimulateMeans()`. 
+#' @param params splatPopParams object containing parameters for population
+#'        scale simulations. See \code{\link{splatPopParams}} for details. 
+#'        Default=`newSplatPopParams()`.
+#' @param method which simulation method to use. Options are "single" which
+#'        produces a single cell population for each sample, "groups" which 
+#'        produces distinct groups (eg. cell types) for each sample (note, this
+#'        creates separate groups from those created in `popSimulate` with only
+#'        DE effects), and "paths" which selects cells from continuous
+#'        trajectories (eg. differentiation processes).
+#' @param counts_only logical. Whether to return only the counts (Default=FALSE).
+#' @param verbose logical. Whether to print progress messages.
+#' @param ... any additional parameter settings to override what is provided in
+#'        \code{params}.
+#'
+#' @return SingleCellExperiment with simulated data for whole population.
+#'
+#' @importFrom SingleCellExperiment SingleCellExperiment cbind
+#' @importFrom SummarizedExperiment rowData rowData<-
+#' @export
+#' 
+splatPopSimulateSC <- function(sim_means,
```

Comment:
I wonder if there might be better names for this and `splatPopSimulateMeans`? I used `kersplatSetup()` and `kersplatSample()` but maybe not appropriate here. I'll have a think but not super important.

Reply:
I went with these names because it clearly distinguishes between the two phases of simulations in splatPop. Happy to hear other suggestions!